libs/auth/src/inserted_smart_card_auth_with_memory_card.ts

```diff
@@ -232,11 +236,19 @@ export class InsertedSmartCardAuthWithMemoryCard
               if (validationResult.isOk()) {
                 assert(user);
                 if (currentAuthStatus.status === 'logged_out') {
-                  if (
-                    user.role === 'system_administrator' ||
-                    user.role === 'election_manager'
-                  ) {
-                    return { status: 'checking_passcode', user };
+                  if (user.role === 'system_administrator') {
+                    return isFeatureFlagEnabled(
+                      BooleanEnvironmentVariableName.SKIP_PIN_ENTRY
+                    )
+                      ? { status: 'logged_in', user }
+                      : { status: 'checking_passcode', user };
+                  }
+                  if (user.role === 'election_manager') {
+                    return isFeatureFlagEnabled(
+                      BooleanEnvironmentVariableName.SKIP_PIN_ENTRY
+                    )
+                      ? { status: 'logged_in', user }
+                      : { status: 'checking_passcode', user };
```

Comment:
Even though the logic is identical for sys admins and election managers, when I group everything under one case, e.g.

```ts
if (user.role === 'system_administrator' || user.role === 'election_manager') {
  return isFeatureFlagEnabled(
    BooleanEnvironmentVariableName.SKIP_PIN_ENTRY
  )
    ? { status: 'logged_in', user }
    : { status: 'checking_passcode', user };
}
```

I get the following type error:

<img width="885" alt="type-error" src="https://user-images.githubusercontent.com/12616928/220733536-c460fb0b-0599-404e-a62b-115f9469caa1.png">

Hence the separate cases when it seems like I could just use one